arangod/IResearch/IResearchFeature.cpp

```diff
@@ -166,39 +167,53 @@ size_t computeThreadPoolSize(size_t threads, size_t threadsLimit) {
                                      size_t(std::thread::hardware_concurrency()) / 4));
 }
 
-bool iresearchViewUpgradeVersion0_1(TRI_vocbase_t& vocbase,
-                                    arangodb::velocypack::Slice const& upgradeParams) {
-  std::vector<std::shared_ptr<arangodb::LogicalView>> views;
-
-  if (arangodb::ServerState::instance()->isCoordinator()) {
-    auto* ci = arangodb::ClusterInfo::instance();
-
-    if (!ci) {
-      LOG_TOPIC("1804b", WARN, arangodb::iresearch::TOPIC)
-          << "failure to find 'ClusterInfo' instance while upgrading "
-             "IResearchView from version 0 to version 1";
+bool upgradeSignleServerArangoSearchView0_1(
+    TRI_vocbase_t& vocbase,
+    arangodb::velocypack::Slice const& /*upgradeParams*/) {
+  using arangodb::application_features::ApplicationServer;
+
+  // NOTE: during the upgrade 'ClusterFeature' is disabled which means 'ClusterFeature::validateOptions(...)'
+  // hasn't been called and server role in 'ServerState' is not set properly.
+  // In order to upgrade ArangoSearch views from version 0 to version 1 we need to
+  // differentiate between signle server and cluster, therefore we temporary set role in 'ServerState',
+  // actually supplied by a user, only for the duration of task to avoid other upgrade tasks, that
+  // potentially rely on the original behaviour, to be affected.
+  struct ServerRoleGuard {
+    ServerRoleGuard() {
+      auto const* clusterFeature = ApplicationServer::lookupFeature<arangodb::ClusterFeature>("Cluster");
+      auto* state = arangodb::ServerState::instance();
+
+      if (state && clusterFeature && !clusterFeature->isEnabled()) {
+        auto const role = arangodb::ServerState::stringToRole(clusterFeature->myRole());
+
+        // only for cluster
+        if (role > arangodb::ServerState::ROLE_SINGLE) {
```

Comment:
this will include Agents too. Is this intended?

Reply:
If not, then using something like `ServerState::isClusterrole(role)` may be more intuitive.

Reply:
done